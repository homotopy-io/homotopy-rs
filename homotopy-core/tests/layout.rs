use homotopy_core::{examples, layout::Layout};
use insta::assert_debug_snapshot;

#[test]
fn monoid_unit() {
    let (_, diagram) = examples::monoid_unit();

    let layout = Layout::<2>::new(&diagram.into()).unwrap();

    assert_debug_snapshot!(layout.positions);
}

#[test]
fn scalar_and_beads() {
    let (_, diagram) = examples::scalar_and_beads();

    let layout = Layout::<2>::new(&diagram.into()).unwrap();

    assert_debug_snapshot!(layout.positions);
}

#[test]
fn associator() {
    let (_, diagram) = examples::associator();

    let layout = Layout::<3>::new(&diagram.into()).unwrap();

    assert_debug_snapshot!(layout.positions);
}

#[test]
fn snake() {
    let (_, diagram) = examples::algebraic_snake();

    let layout = Layout::<3>::new(&diagram.into()).unwrap();

    assert_debug_snapshot!(layout.positions);
}

#[test]
fn lips() {
    let (_, diagram) = examples::lips();

    let layout = Layout::<4>::new(&diagram.into()).unwrap();

    assert_debug_snapshot!(layout.positions);
}

#[test]
fn pants_unit() {
    let (_, diagram) = examples::pants_unit();

    let layout = Layout::<4>::new(&diagram.into()).unwrap();

    assert_debug_snapshot!(layout.positions);
}
