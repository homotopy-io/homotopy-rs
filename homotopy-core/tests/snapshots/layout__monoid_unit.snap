---
source: homotopy-core/tests/layout.rs
expression: layout.positions
---
{
    [
        Boundary(
            Target,
        ),
        Boundary(
            Source,
        ),
    ]: [
        8.0,
        0.0,
    ],
    [
        Boundary(
            Target,
        ),
        Interior(
            Regular(
                0,
            ),
        ),
    ]: [
        8.0,
        2.0,
    ],
    [
        Interior(
            Regular(
                0,
            ),
        ),
        Interior(
            Regular(
                0,
            ),
        ),
    ]: [
        1.0,
        1.0,
    ],
    [
        Interior(
            Singular(
                1,
            ),
        ),
        Interior(
            Singular(
                1,
            ),
        ),
    ]: [
        4.0,
        5.0,
    ],
    [
        Interior(
            Regular(
                0,
            ),
        ),
        Boundary(
            Source,
        ),
    ]: [
        1.0,
        0.0,
    ],
    [
        Interior(
            Regular(
                1,
            ),
        ),
        Boundary(
            Target,
        ),
    ]: [
        3.0,
        7.0,
    ],
    [
        Interior(
            Singular(
                2,
            ),
        ),
        Boundary(
            Target,
        ),
    ]: [
        6.0,
        7.0,
    ],
    [
        Boundary(
            Source,
        ),
        Boundary(
            Source,
        ),
    ]: [
        0.0,
        0.0,
    ],
    [
        Interior(
            Regular(
                2,
            ),
        ),
        Interior(
            Regular(
                2,
            ),
        ),
    ]: [
        5.0,
        6.0,
    ],
    [
        Boundary(
            Target,
        ),
        Interior(
            Singular(
                0,
            ),
        ),
    ]: [
        8.0,
        4.0,
    ],
    [
        Interior(
            Singular(
                0,
            ),
        ),
        Boundary(
            Target,
        ),
    ]: [
        2.0,
        7.0,
    ],
    [
        Interior(
            Regular(
                1,
            ),
        ),
        Interior(
            Singular(
                0,
            ),
        ),
    ]: [
        3.0,
        3.0,
    ],
    [
        Interior(
            Regular(
                2,
            ),
        ),
        Interior(
            Singular(
                0,
            ),
        ),
    ]: [
        5.0,
        3.0,
    ],
    [
        Interior(
            Regular(
                2,
            ),
        ),
        Boundary(
            Target,
        ),
    ]: [
        5.0,
        7.0,
    ],
    [
        Interior(
            Regular(
                0,
            ),
        ),
        Interior(
            Singular(
                0,
            ),
        ),
    ]: [
        1.0,
        2.0,
    ],
    [
        Boundary(
            Target,
        ),
        Interior(
            Regular(
                1,
            ),
        ),
    ]: [
        8.0,
        5.5,
    ],
    [
        Interior(
            Singular(
                1,
            ),
        ),
        Interior(
            Regular(
                1,
            ),
        ),
    ]: [
        4.0,
        4.0,
    ],
    [
        Interior(
            Singular(
                1,
            ),
        ),
        Interior(
            Regular(
                2,
            ),
        ),
    ]: [
        4.0,
        6.0,
    ],
    [
        Boundary(
            Source,
        ),
        Interior(
            Singular(
                1,
            ),
        ),
    ]: [
        0.0,
        4.0,
    ],
    [
        Interior(
            Regular(
                0,
            ),
        ),
        Interior(
            Regular(
                2,
            ),
        ),
    ]: [
        1.0,
        5.5,
    ],
    [
        Interior(
            Singular(
                0,
            ),
        ),
        Interior(
            Singular(
                0,
            ),
        ),
    ]: [
        2.0,
        3.0,
    ],
    [
        Interior(
            Regular(
                1,
            ),
        ),
        Boundary(
            Source,
        ),
    ]: [
        3.0,
        0.0,
    ],
    [
        Interior(
            Singular(
                1,
            ),
        ),
        Boundary(
            Source,
        ),
    ]: [
        4.0,
        0.0,
    ],
    [
        Boundary(
            Source,
        ),
        Boundary(
            Target,
        ),
    ]: [
        0.0,
        7.0,
    ],
    [
        Interior(
            Regular(
                1,
            ),
        ),
        Interior(
            Regular(
                0,
            ),
        ),
    ]: [
        3.0,
        1.5,
    ],
    [
        Interior(
            Regular(
                2,
            ),
        ),
        Interior(
            Regular(
                0,
            ),
        ),
    ]: [
        5.0,
        1.5,
    ],
    [
        Interior(
            Regular(
                3,
            ),
        ),
        Boundary(
            Source,
        ),
    ]: [
        7.0,
        0.0,
    ],
    [
        Boundary(
            Source,
        ),
        Interior(
            Regular(
                2,
            ),
        ),
    ]: [
        0.0,
        5.5,
    ],
    [
        Boundary(
            Source,
        ),
        Interior(
            Regular(
                1,
            ),
        ),
    ]: [
        0.0,
        3.0,
    ],
    [
        Interior(
            Regular(
                2,
            ),
        ),
        Interior(
            Singular(
                1,
            ),
        ),
    ]: [
        5.0,
        5.0,
    ],
    [
        Interior(
            Regular(
                3,
            ),
        ),
        Interior(
            Regular(
                0,
            ),
        ),
    ]: [
        7.0,
        2.0,
    ],
    [
        Boundary(
            Target,
        ),
        Boundary(
            Target,
        ),
    ]: [
        8.0,
        7.0,
    ],
    [
        Interior(
            Singular(
                2,
            ),
        ),
        Interior(
            Regular(
                1,
            ),
        ),
    ]: [
        6.0,
        6.0,
    ],
    [
        Interior(
            Singular(
                0,
            ),
        ),
        Interior(
            Regular(
                0,
            ),
        ),
    ]: [
        2.0,
        1.0,
    ],
    [
        Boundary(
            Source,
        ),
        Interior(
            Singular(
                0,
            ),
        ),
    ]: [
        0.0,
        2.0,
    ],
    [
        Interior(
            Regular(
                0,
            ),
        ),
        Interior(
            Regular(
                1,
            ),
        ),
    ]: [
        1.0,
        3.0,
    ],
    [
        Interior(
            Singular(
                2,
            ),
        ),
        Interior(
            Singular(
                0,
            ),
        ),
    ]: [
        6.0,
        4.0,
    ],
    [
        Interior(
            Singular(
                1,
            ),
        ),
        Interior(
            Singular(
                0,
            ),
        ),
    ]: [
        4.0,
        3.0,
    ],
    [
        Interior(
            Singular(
                1,
            ),
        ),
        Boundary(
            Target,
        ),
    ]: [
        4.0,
        7.0,
    ],
    [
        Interior(
            Regular(
                0,
            ),
        ),
        Boundary(
            Target,
        ),
    ]: [
        1.0,
        7.0,
    ],
    [
        Interior(
            Singular(
                2,
            ),
        ),
        Boundary(
            Source,
        ),
    ]: [
        6.0,
        0.0,
    ],
    [
        Interior(
            Regular(
                3,
            ),
        ),
        Interior(
            Singular(
                0,
            ),
        ),
    ]: [
        7.0,
        4.0,
    ],
    [
        Interior(
            Regular(
                1,
            ),
        ),
        Interior(
            Regular(
                1,
            ),
        ),
    ]: [
        3.0,
        5.0,
    ],
    [
        Interior(
            Regular(
                2,
            ),
        ),
        Interior(
            Regular(
                1,
            ),
        ),
    ]: [
        5.0,
        4.0,
    ],
    [
        Interior(
            Regular(
                3,
            ),
        ),
        Interior(
            Regular(
                1,
            ),
        ),
    ]: [
        7.0,
        5.5,
    ],
    [
        Interior(
            Singular(
                0,
            ),
        ),
        Boundary(
            Source,
        ),
    ]: [
        2.0,
        0.0,
    ],
    [
        Boundary(
            Source,
        ),
        Interior(
            Regular(
                0,
            ),
        ),
    ]: [
        0.0,
        1.0,
    ],
    [
        Interior(
            Regular(
                2,
            ),
        ),
        Boundary(
            Source,
        ),
    ]: [
        5.0,
        0.0,
    ],
    [
        Interior(
            Regular(
                3,
            ),
        ),
        Boundary(
            Target,
        ),
    ]: [
        7.0,
        7.0,
    ],
    [
        Interior(
            Regular(
                0,
            ),
        ),
        Interior(
            Singular(
                1,
            ),
        ),
    ]: [
        1.0,
        4.0,
    ],
    [
        Interior(
            Singular(
                2,
            ),
        ),
        Interior(
            Regular(
                0,
            ),
        ),
    ]: [
        6.0,
        1.5,
    ],
    [
        Interior(
            Singular(
                1,
            ),
        ),
        Interior(
            Regular(
                0,
            ),
        ),
    ]: [
        4.0,
        1.5,
    ],
    [
        Interior(
            Singular(
                0,
            ),
        ),
        Interior(
            Regular(
                1,
            ),
        ),
    ]: [
        2.0,
        5.5,
    ],
}
